webrender/src/device/gfx/device.rs

```diff
@@ -1817,58 +1767,32 @@ impl<B: hal::Backend> Device<B> {
                 }
             },
             Entry::Vacant(v) => {
-                let per_frame_descriptor_sets = self.per_frame_descriptor_sets.get_mut(&descriptor_group).unwrap();
-                if per_frame_descriptor_sets.is_empty() {
+                if descriptor_sets.is_empty() {
                     unsafe {
-                        self.desc_allocator.allocate(
-                            &self.device,
-                            &self.descriptor_set_layouts[&descriptor_group][DESCRIPTOR_SET_PER_FRAME],
-                            self.descriptor_set_ranges[&descriptor_group][DESCRIPTOR_SET_PER_FRAME],
+                        desc_allocator.allocate(
+                            device,
+                            descriptor_layout,
+                            descriptor_ranges,
                             DESCRIPTOR_COUNT,
-                            per_frame_descriptor_sets,
+                            descriptor_sets,
                         )
                     }.expect("Allocate descriptor sets failed");
                 }
-                let desc_set = v.insert(per_frame_descriptor_sets.pop().unwrap());
-                for &(index, sampler_name) in SAMPLERS[PER_DRAW_SAMPLER_COUNT..].iter() {
-                    program.bound_textures[index] = self.bound_textures[index];
-                    let image = &self.images[&self.bound_textures[index]].core;
+                let desc_set = v.insert(descriptor_sets.pop().unwrap());
+                for &(index, sampler_name) in SAMPLERS[range].iter() {
+                    program.bound_textures[index] = bound_textures[index];
```

Comment:
why are we still tracking textures per program? This was required for GL, but it's not like the binding model works in Vulkan anyway